File: leetcode/group2/504.py
class Solution:
    def convertToBase7(self, num: int) -> str:
        if num == 0:
            return str(num)
        abs_num = abs(num)
        ans = []

        while abs_num:
            ans = [str(abs_num % 7)] + ans
            abs_num //= 7
        ret = ''.join(ans)
        return '-'+ret if num < 0 else ret

File: leetcode/花花酱题集/504.js
/**
 * @param {number} num
 * @return {string}
 */
var convertToBase7 = function (num) {
  if (num === 0) {
    return num.toString();
  }

  const negative = num > 0 ? false : true;
  let numAbs = Math.abs(num);
  let ans = "";

  while (numAbs > 0) {
    ans = (numAbs % 7) + ans;
    numAbs = parseInt(numAbs / 7);
  }
  return negative ? "-" + ans : ans;
};

console.log(convertToBase7(0));
